File: packages/jexlang-java/src/main/java/com/jexlang/java/eval/errors/UndefinedTransformError.java
package com.jexlang.java.eval.errors;

public class UndefinedTransformError extends JexLangRuntimeError {
    private final String transformName;

    public UndefinedTransformError(String transformName) {
        super("Undefined transform: " + transformName);
        this.transformName = transformName;
    }

    public String getTransformName() {
        return transformName;
    }

    @Override
    public String getName() {
        return "UndefinedTransformError";
    }
}

File: packages/jexlang-swift/jexlang-swift/eval/errors/UndefinedTransformError.swift
//
//  UndefinedTransformError.swift
//  jexlang-swift
//
//  Created by Vinay Kumar on 03/10/25.
//

import Foundation

public class UndefinedTransformError: JexLangRuntimeError {
    public let transformName: String
    
    init(transformName: String) {
        self.transformName = transformName
        super.init(message: "Undefined transform: " + transformName)
    }
    
    public func getTransformName() -> String {
        return self.transformName;
    }
}
